rust/basic_bitcoin/src/basic_bitcoin/src/bitcoin_wallet.rs

```diff
@@ -0,0 +1,325 @@
+pub async fn send(
+    network: Network,
+    derivation_path: Vec<Vec<u8>>,
+    dst_address: String,
+    amount: Satoshi,
+) {
+    // Get fee percentiles from previous transactions to estimate our own fee.
+    let fee_percentiles = bitcoin_api::get_current_fee_percentiles(network).await;
+
+    let fee_per_byte = if fee_percentiles.is_empty() {
+        // There are no fee percentiles. This case can only happen on a regtest
+        // network where there are no non-coinbase transactions. In this case,
+        // we use a default of 1000 millisatoshis/byte (i.e. 1 satoshi/byte)
+        1000
+    } else {
+        // Choose the 75th percentile for sending fees so that the transaction
+        // is mined relatively quickly.
+        fee_percentiles[74]
+    };
+
+    // Fetch our public key, P2PKH address, and UTXOs.
+    let own_public_key = ecdsa_api::ecdsa_public_key(derivation_path.clone()).await;
+    let own_address = public_key_to_p2pkh_address(network, &own_public_key);
+    let own_utxos = bitcoin_api::get_utxos(network, own_address.clone())
+        .await
+        .utxos;
+
+    let own_address = Address::from_str(&own_address).unwrap();
+    let dst_address = Address::from_str(&dst_address).unwrap();
+
+    // Build the transaction that sends `amount` to the destination address.
+    let transaction = build_transaction(
+        &own_public_key,
+        &own_address,
+        &own_utxos,
+        &dst_address,
+        amount,
+        fee_per_byte,
+    )
+    .await
+    .expect("Error building transaction.");
+
+    let tx_bytes = transaction.serialize();
+    print(&format!("Transaction to sign: {}", hex::encode(tx_bytes)));
+
+    // Sign the transaction.
+    let signed_transaction = sign_transaction(
+        &own_public_key,
+        &own_address,
+        transaction,
+        derivation_path,
+        ecdsa_api::sign_with_ecdsa,
+    )
+    .await;
+
+    let signed_transaction_bytes = signed_transaction.serialize();
+    print(&format!(
+        "Signed transaction: {}",
+        hex::encode(&signed_transaction_bytes)
+    ));
+
+    print("Sending transaction...");
+    bitcoin_api::send_transaction(network, signed_transaction_bytes).await;
+    print("Done");
+}
+
+// Builds a transaction to send the given `amount` of satoshis to the
+// destination address.
+async fn build_transaction(
+    own_public_key: &[u8],
+    own_address: &Address,
+    own_utxos: &[Utxo],
+    dst_address: &Address,
+    amount: u64,
+    fee_per_byte: MillisatoshiPerByte,
+) -> Result<Transaction, String> {
+    // We have a chicken-and-egg problem where we need to know the length
+    // of the transaction in order to compute its proper fee, but we need
+    // to know the proper fee in order to figure out the inputs needed for
+    // the transaction.
+    //
+    // We solve this problem iteratively. We start with a fee of zero, build
+    // and sign a transaction, see what it's size is, and then update then
```

Comment:
The suggestion button is grayed out (?), so:
it's size -> its size
then update then update -> then update